revan-hameed-dunia/src/components/header.js

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from 'react'
+import {Navbar, Nav, Form, FormControl, Button, Spinner} from 'react-bootstrap'
+import fetchData from './fetchData'
+import Category from './category'
+
+export default function Header(props) {
+  const [data, setData] = useState([])
+  const constructUrl = (path, query) => {
+    const TMDB_BASE_URL = "https://api.themoviedb.org/3";
+
+    return `${TMDB_BASE_URL}/${path}?api_key=${atob(
+      "ZDJmYTdhZDFlMjZhZjA4NDdkMzQ5ZDdkYmQ1ZjkzZTU="
+    )}&query=${query}`;;
+}
+const trending =
+  "https://api.themoviedb.org/3/trending/all/day?api_key=754ad3989128c7d8cfcc82e6591e7f2e";
```

Comment:
I know the construct URL is was only optimized for the search box and was not required to be used here but just for info, best practice would be to optimize the construct URL function to construct the URL and making sure the function would construct the URL whether it has a query or not.

Even though, you might think that it's overkill, it really is not, as soon as you get the function to construct URLs correctly, you wouldn't need to worry about the structure of URLs for other areas in your web app.